# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.cpp
# Commit: 2c4f84b0e661
# Full Hash: 2c4f84b0e66164503066908c973fef983623c864
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2020-01-14 03:45:02
# Regressor Bug: 1599043
# File Overlap Count: 1
# Description:
#   Bug 1599043 - Part 1: Add a ContentBlockingLog into the WindowGlobalParent r=dimi,Ehsan
#   
#   This adds a ContentBlockingLog into the WindowGlobalParent. This
#   ContentBlockingLog is bascially a copy of the log in the content
#   process. This log in parent is needed for the OnContentBlockingEvent in
# ==============================================================================

diff -r b54a7ff4c097 -r 2c4f84b0e661 dom/ipc/WindowGlobalParent.cpp
--- a/dom/ipc/WindowGlobalParent.cpp	Mon Jan 13 14:07:09 2020 +0000
+++ b/dom/ipc/WindowGlobalParent.cpp	Mon Jan 13 14:06:50 2020 +0000
@@ -291,6 +291,35 @@
   return VoidString();
 }
 
+void WindowGlobalParent::NotifyContentBlockingEvent(
+    uint32_t aEvent, nsIRequest* aRequest, bool aBlocked, nsIURI* aURIHint,
+    const nsTArray<nsCString>& aTrackingFullHashes,
+    const Maybe<AntiTrackingCommon::StorageAccessGrantedReason>& aReason) {
+  MOZ_ASSERT(NS_IsMainThread());
+  MOZ_ASSERT(aURIHint);
+  DebugOnly<bool> isCookiesBlockedTracker =
+      aEvent == nsIWebProgressListener::STATE_COOKIES_BLOCKED_TRACKER ||
+      aEvent == nsIWebProgressListener::STATE_COOKIES_BLOCKED_SOCIALTRACKER;
+  MOZ_ASSERT_IF(aBlocked, aReason.isNothing());
+  MOZ_ASSERT_IF(!isCookiesBlockedTracker, aReason.isNothing());
+  MOZ_ASSERT_IF(isCookiesBlockedTracker && !aBlocked, aReason.isSome());
+
+  // Return early if this WindowGlobalParent is in process.
+  if (IsInProcess()) {
+    return;
+  }
+
+  nsAutoCString origin;
+  nsContentUtils::GetASCIIOrigin(aURIHint, origin);
+
+  Maybe<uint32_t> event = GetContentBlockingLog()->RecordLogParent(
+      origin, aEvent, aBlocked, aReason, aTrackingFullHashes);
+
+  if (event) {
+    // Notify the OnContentBlockingEvent.
+  }
+}
+
 already_AddRefed<JSWindowActorParent> WindowGlobalParent::GetActor(
     const nsAString& aName, ErrorResult& aRv) {
   if (!CanSend()) {